tests/codebuild/create_tests.sh

```diff
@@ -139,12 +139,12 @@ function run_hap_test()
   local hostingdeployment_type="hostingdeployment"
 
   # Create the second Endpoint
-  sed -i "s/$hosting_deployment_1/$hosting_deployment_2/g" testfiles/xgboost-hosting-deployment.yaml
+  yq w -i testfiles/xgboost-hosting-deployment.yaml "metadata.name" $hosting_deployment_2
   run_test "${crd_namespace}" testfiles/xgboost-hosting-deployment.yaml
```

Comment:
Instead of modifying the test file inline (`yq w -i`), can we just write it to a new file and then apply that?

Reply:
yq is cleaner in IMO, why do you prefer copying files ?

Reply:
I mean instead of using `yq w -i ...` we can do `yq w ... testfiles/xgboost-hosting-deployment-2.yaml` then `run_test ${crd_namespace} testfiles/xgboost-hosting-deployment-2.yaml`. This way we are sure that every time we modify the file, we are modifying the same input file.

Reply:
sorry i didn’t see this message before merging.

i feel we should work on new yamls by default for all the tests and never touch originals 
even when injecting variables here : [inject_tests.sh](https://github.com/aws/amazon-sagemaker-operator-for-k8s/blob/master/tests/codebuild/inject_tests.sh)

but for now yq is fine though